### PR TITLE
CTW-673/Medications Sort Button

### PR DESCRIPTION
**Pull request by mrosata**
- Meds filtering and sorting use hooks like conditions
- Generic sort button _(mostly the same as the button created for conditions but it uses the sort indices pattern used by Table component)_
- Add sort to Medications component

https://www.loom.com/share/713120fea71e4a38a15f7f89ac607c20

### CODE REVIEW

File: .storybook/preview.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 @import "../src/components/core/main.scss";
 
-body {
+body * {
```
Comment (mrosata) [inline .storybook/preview.scss:3]: This makes it so `ctw-font-medium` and `ctw-font-light` show up as different font weights.

File: src/components/content/medications-table-base.tsx [hunk 2]
```diff
@@ -111,10 +111,11 @@ export const MedicationsTableBase = ({
     >
       <div className="ctw-overflow-hidden">
         <Table
+          sort={!breakpoints.sm ? sort : undefined}
+          onSort={!breakpoints.sm ? setSort : undefined}
+          showTableHead={!breakpoints.sm}
```
Comment (mrosata) [inline src/components/content/medications-table-base.tsx:116]: Tai suggested only showing sort on small breakpoint.

File: src/components/content/medications/patient-medications-tabbed.tsx [hunk 3]
```diff
@@ -78,68 +76,40 @@ export function OtherProviderMedsTableTab({
...
-    !otherProviderMedications || prescriberNames.length < 2
-      ? null
-      : {
-          key: "providers",
-          type: "checkbox",
-          icon: "clipboard",
-          values: prescriberNames,
-          display: "prescriber",
-        },
-  ]);
```
Comment (mrosata) [inline src/components/content/medications/patient-medications-tabbed.tsx:132]: moved this logic into patient-medications-filters.tsx to line up more with how conditions does filter and sort

File: src/components/core/sort-button/sort-button.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      name: option.display,
+      isSelected: currentSorts.display === option.display,
+    }))}
+  >
+    <div className="ctw-btn-default ctw-flex ctw-items-center ctw-space-x-1">
+      <span>Sort</span>
+      <ChevronDownIcon className="ctw-h-4" />
+    </div>
+  </DropdownMenuAction>
+);
```
Comment (mrosata) [inline src/components/core/sort-button/sort-button.tsx:46]: @WesleyKapow

**Review by WesleyKapow — APPROVED** [review]
LGTM

I've got a few tweaks in addition to your sort-button but I'll add those to yours after you merge this in.

**Pull request merged**